test/cli/cli_test.cpp

```diff
@@ -45,9 +46,33 @@ struct cli_test : public ::testing::Test
 
     // Generate the full path of a test file that is provided in the data directory.
     static
-    std::string data(std::string const & filename)
+    std::filesystem::path data(std::string const & filename)
     {
-        return std::string{DATADIR}.append(filename);
+        return std::filesystem::path{std::string{DATADIR}}.concat(filename);
+    }
+
+    // Create a unique work directory for a test and return the full path for a given file name.
+    static
+    std::filesystem::path workdir(std::string const & filename)
+    {
+        // Derive unique test directory name from the test name.
```

Comment:
this is not really unique, right?

What happens if I rerun the test? I don't find where the created files get deleted.

Reply:
> this is not really unique, right?

It is unique because two tests cannot have the same name... But maybe I misunderstand your question.

> What happens if I rerun the test? I don't find where the created files get deleted.

Existing files are overwritten with the latest test result. In my opinion it does not make sense to keep (identical) older versions which were generated with the same test code. I was planning to add the deletion of the output directory contents to the `make clean` target.

Reply:
What happens with apps that don't overwrite files (like we discussed)

Reply:
Hm hm true... maybe I implement that the directory contents are deleted before the test runs.